src/components/Map/index.tsx

```diff
@@ -158,59 +84,55 @@ const Map = () => {
     )
       return;
 
-    const geocoder = new kakao.maps.services.Geocoder();
-
-    cityDustInfoList.forEach(
-      ({
-        location,
-        fineDustScale,
-        fineDustGrade,
-        ultraFineDustScale,
-        ultraFineDustGrade,
-      }) => {
-        geocoder.addressSearch(location, (result, status) => {
-          if (status === kakao.maps.services.Status.OK) {
-            const latitude = Number(result[0].y);
-            const longitude = Number(result[0].x);
-
-            const template = makeMarkerTemplate({
-              location,
-              fineDustScale,
-              fineDustGrade,
-              ultraFineDustScale,
-              ultraFineDustGrade,
-            });
+    makeCityMarker({
+      map,
+      markers: cityDustInfoMarkers,
+      dustInfoList: cityDustInfoList,
+      setCityDustInfoMarkers,
+    });
 
-            const marker = new kakao.maps.CustomOverlay({
-              map,
-              position: new kakao.maps.LatLng(latitude, longitude),
-              content: template,
-            });
+    return () => {
+      removeMarker({ map, markers: cityDustInfoMarkers });
+    };
+  }, [cityDustInfoList]);
 
-            if (
-              !cityDustInfoMarkers.find(
-                (value) => value.getPosition() === marker.getPosition()
-              )
-            )
-              setCityDustInfoMarkers((prev) => [...prev, marker]);
-          }
-        });
-      }
-    );
+  // 마커에 이벤트 추가, 제거
+  useEffect(() => {
+    document
+      .querySelectorAll<HTMLDivElement>('.dust-info-marker')
+      .forEach((city) => {
+        city.onclick = () => handleClickMarker(city);
+        city.onmouseover = () => handleMouseOverMarker(city);
+        city.onmouseout = () => handleMouseOutMarker(city);
+      });
 
     return () => {
-      setMakerToNull({ map, markers: cityDustInfoMarkers });
+      document
+        .querySelectorAll<HTMLDivElement>('.dust-info-marker')
+        .forEach((city) => {
+          city.removeEventListener('click', () => handleClickMarker(city));
+          city.removeEventListener('mouseover', () =>
+            handleMouseOverMarker(city)
+          );
+          city.removeEventListener('mouseout', () =>
+            handleMouseOutMarker(city)
+          );
+        });
     };
-  }, [cityDustInfoList]);
+  }, [currentLocation, cityDustInfoMarkers, zoomLevel]);
 
   const handleClickMarker = useCallback((city: HTMLDivElement) => {
-    setCity(city.id);
+    setSelectedSidoOrCity(city.id);
 
     const nextDustInfo = {
-      fineDustScale: Number(city.dataset.finedustscale || 1),
-      fineDustGrade: Number(city.dataset.finedustgrade || 1),
-      ultraFineDustScale: Number(city.dataset.ultrafinedustscale || 1),
-      ultraFineDustGrade: Number(city.dataset.ultrafinedustgrade || 1),
+      fineDustScale: Number(city.dataset.finedustscale || INIT_DUST_SCALE),
```

Comment:
```suggestion
      fineDustScale: Number(city.dataset.finedustscale) || INIT_DUST_SCALE,
```

요렇게 작성할 수도 있을 것 같아요~!